### PR TITLE
Theo/add jwt auth helper mixin

### PR DESCRIPTION
**Pull request by tnsengimana**
The main change is an addition of AuthenticationMixin class which will help in writing tests that hit a service endpoints. It provides some methods to switch between different jwt tokens. There 2 pending pull requests ([geolocation](https://github.com/ZeroCater/mp-geolocation/pull/22) & [slots-and-orders](https://github.com/ZeroCater/mp-slots-and-orders/pull/32)) that depend on this new class.

### CODE REVIEW

File: zc_common/jwt_auth/tests.py [hunk 1]
```diff
@@ -50,3 +53,38 @@ def get_service_endpoint_urls(urlconfig=None, default_value='1'):
...
+        staff_user = self.create_user(['user', 'staff'], user_id)
+        return self.create_user_token(staff_user)
+
+    def get_user_token(self, user_id=1):
+        user = self.create_user(['user'], user_id)
+        return self.create_user_token(user)
+
+    def get_guest_token(self, user_id=1):
+        user = self.create_user(['user'], user_id)
+        setattr(user, 'isGuest', True)
```
Comment (birkholz) [inline zc_common/jwt_auth/tests.py]: This is incorrect. Anonymous users have roles `['guest']`, and there should be a function here for authenticating as such.

For "Guest" users, there is no `user.isGuest` attribute. They're just like normal users.

**Comment by hamedahmadi** [comment]
@birkholz This PR has some breaking changes, can we merge this in few days? I don't want to deal with fixing these in the repos for the next few days.

**Comment by hamedahmadi** [comment]
@birkholz If you are blocked by this, please go ahead and merge it

**Comment by birkholz** [comment]
@hamedahmadi I'm not using these changes at all.

**Comment by tnsengimana** [comment]
@hamedahmadi I have removed those commits. It shouldn't break anything.

**Comment by tnsengimana** [comment]
@birkholz it's blocking both mp-slots-and-orders & mp-geolocation. I mentioned this in the comment above.

File: zc_common/jwt_auth/tests.py (hunk 1 above)
Comment (amberylx) [inline zc_common/jwt_auth/tests.py]: I think there was a misunderstanding due to overloading of the term 'guest'. Guest users having role of 'user', but Anonymous users having the role of 'guest' is really confusing.

**Comment by hamedahmadi** [comment]
> I think there was a misunderstanding due to overloading of the term 'guest'. Guest users having role of 'user', but Anonymous users having the role of 'guest' is really confusing.

@birkholz @tnsengimana I think @amberylx has a valid point, can we think of a better term for anonymous users? maybe use `anon` instead?

**Comment by tnsengimana** [comment]
@hamedahmadi @amberylx I am totally lost now! I thought "anonymous users" are basically "guest users". What does this really mean?
> Guest users having role of 'user', but Anonymous users having the role of 'guest'...

**Comment by amberylx** [comment]
@hamedahmadi I like that much better. @tnsengimana I'm happy to share what I know about anonymous users, guest users, and roles, feel free to slack me if you have questions after discussing with @hamedahmadi.

**Comment by tnsengimana** [comment]
Great, sounds like the agreement is to rename the anonymous users' role. I made the change to `anonymous`. I think it's more clearer than `anon`, altho a little longer.

File: zc_common/jwt_auth/tests.py (hunk 1 above)
Comment (tnsengimana) [inline zc_common/jwt_auth/tests.py]: @birkholz I synced up with @hamedahmadi about `user.isGuest` and he confirmed the flag is actually needed to make a distinction with regular users.

Comment (birkholz) [inline zc_common/jwt_auth/tests.py]: This is in a test though. `isGuest` is never set during JWT creation by the User service.
So the tests are not matching reality.

**Comment by birkholz** [comment]
I'm fine with renaming the "guest" role to "anonymous", but the change will need to be propagated to all places where "guest" is expected, and the gateway's JWT will need to be replaced as well.

**Comment by tnsengimana** [comment]
@birkholz that was my concern actually. Should we postpone the change? I think it's a lower priority issue.

**Comment by birkholz** [comment]
👍

**Pull request merged**